src/gorilla.c

```diff
@@ -442,14 +442,17 @@ static double readV(Compressed_Iterator *iter) {
 
 ChunkResult Compressed_ReadNext(Compressed_Iterator *iter, timestamp_t *timestamp, double *value) {
     assert(iter);
-    *timestamp = iter->prevTS;
-    *value     = iter->prevValue.d;
-
     assert(iter->chunk);
-    if (iter->count >= iter->chunk->count) return CR_END;
 
-    iter->prevTS      = readTS(iter);
-    iter->prevValue.d = readV (iter);
+    if (iter->count >= iter->chunk->count) return CR_END;
+    
+    if (iter->count == 0) { // First sample
```

Comment:
@ashtul do we have a test that forces specifically the crash on master now and passes in the branch?

Reply:
@filipecosta90 
No. I am not sure if it can be reliably reproduced.
The bug happened because we read data beyond the memory allocated. This caused one of two behaviors. A segmentation fault or an assert failure.
From what I see, the `SIGSEGV` is received only sometimes and my guess is, it depends whether the memory access is allocated to another program or not. In my testing, the same series which read beyond its boundaries didn't always receive the signal.
The assert has a similar story where the error was caused by the data that happened to be in the right after the memory allocation.

Do you have an idea of how to simulate this?